### PR TITLE
reworked docs and made component standards easier to follow/reference

### PR DESCRIPTION
**Pull request by TDH55**

### CODE REVIEW

**Comment by TDH55** [comment]
[Component Standards Docs](https://trello.com/c/bLRBblFp/451-component-standards-docs)

File: website/docs/component-standards/interactions/child-components.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+id: child-components
+title: Child Components
+---
+
+Child components should be used to group related props whenever your component contains props that could be grouped into multiple categories. Child components should:
+
+* Be named in a way that describes their purpose and how the props are related to each other.
+* Be grouped so that each prop in a child component is related in how they interact with the component.
+* Contain an empty state when applicable.
+* Should be able to be enabled and disabled when applicable. See [conditional control](docs/interactions/conditional-control)
```
Comment (adebroux) [inline website/docs/component-standards/interactions/child-components.md:11]:
```suggestion
* Be able to be enabled and disabled when applicable. See [conditional control](docs/interactions/conditional-control)
```

File: website/docs/component-standards/interactions/exporting-data.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Autosave Input
+
+The `autosaveInput` role allows your component to directly modify values in the database.
+
+For more info on `autosaveInput`, see [Modifying Values in the Database](/docs/interactions/modifying-database-values).
+
+## Component Data
+
+Component data allows props from your component to be accessed via magic text in other components, as well as your own. This should be considered the default method for exporting data and should be used any whenever you don't need to directly modify a database value or
```
Comment (adebroux) [inline website/docs/component-standards/interactions/exporting-data.md:16]: Need the rest of this sentence!

File: website/docs/component-standards/interactions/list-controls.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+id: list-controls
+title: List Controls
+---
+
+## Disabling List Controls
+
+There are some situations where you should disable list controls, such as sorting and filtering. In general, any time that a list control either doesn't make sense in the context of you're component or doesn't have any effect on how the component is rendered, then the control should be disabled.
```
Comment (nucks) [inline website/docs/component-standards/interactions/list-controls.md:8]:
```suggestion
There are some situations where you should disable list controls, such as sorting and filtering. In general, any time that a list control either doesn't make sense in the context of your component or doesn't have any effect on how the component is rendered, then the control should be disabled.
```

File: website/docs/component-standards/code/external-apis.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+id: external-apis
+title: External APIs
+---
+
+The use of 3rd party APIs in your components is allowed, but you should be aware of the license and attribution requirements of all APIs you use. 3rd party APIs should also be reputable, well maintained, and not introduce any security vulnerabilities to makers, end users, or the Adalo ecosystem as a whole. If you are unsure of the license or attribution requirements of an API, or whether or not an API is reputable, please contact the Adalo team.
```
Comment (nucks) [inline website/docs/component-standards/code/external-apis.md:6]:
```suggestion
The use of 3rd-party APIs in your components is allowed, but you should be aware of the license and attribution requirements of all APIs you use. 3rd-party APIs should also be reputable, well maintained, and not introduce any security vulnerabilities to makers, end users, or the Adalo ecosystem as a whole. If you are unsure of the license or attribution requirements of an API, or whether or not an API is reputable, please contact the Adalo team.
```

**Review by nucks — CHANGES_REQUESTED** [review]
Left a few comments. Everything else looked good!

File: website/docs/component-standards/interactions/exporting-data.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+For more info on component data, see the [Component Data](/docs/interactions/component-data) docs.
+
+## Action Arguments
+
+Action arguments are very similar to component data, they allow you to use information from your component within an action **in your component only**. One instance where this is useful is when you want to use information from your component in an action, but don't want to share that information across the entire Adalo ecosystem.
```
Comment (adebroux) [inline website/docs/component-standards/interactions/exporting-data.md:22]:
```suggestion
Action arguments are very similar to component data. They allow you to use information from your component within an action **in your component only**. This is useful if you want to use information from your component in an action, but don't want to share that information across the entire Adalo ecosystem.
```

File: website/docs/component-standards/interactions/list-controls.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+id: list-controls
+title: List Controls
+---
+
+## Disabling List Controls
+
+There are some situations where you should disable list controls, such as sorting and filtering. In general, any time that a list control either doesn't make sense in the context of your component or doesn't have any effect on how the component is rendered, then the control should be disabled.
```
Comment (adebroux) [inline website/docs/component-standards/interactions/list-controls.md:8]:
```suggestion
There are some situations where you should disable list controls, such as sorting and filtering. In general, if a list control either doesn't make sense in the context of your component or doesn't have any effect on how the component is rendered, the control should be disabled.
```

File: website/docs/component-standards/component-listing/logo.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---
+id: logo
+title: Logo
+---
+
+The logo in the marketplace is a chance for you to show off your component in a branded way. So while the thumbnail is meant to be a representation of what will actually show up when it’s in the app, the logo is a chance for you to have a little more fun and it’s more of an icon.
```
Comment (adebroux) [inline website/docs/component-standards/component-listing/logo.md:6]:
```suggestion
The logo in the marketplace is a chance for you to show off your component in a branded way. So while the thumbnail is meant to be a representation of what will actually show up in the app, the logo is more of an icon. And it's a chance for you to have a little more fun!
```

File: website/docs/component-standards/code/performance.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+---
+id: performance
+title: Performance
+---
+
+In general, using your component should not lead to reduced performance of a makers app. Some common issues that can cause performance issues include:
+
+- Excessive rendering of components.
+- `console.log` statements
+- Excessive api calls
```
Comment (adebroux) [inline website/docs/component-standards/code/performance.md:10]:
```suggestion
- Excessive API calls
```

**Review by adebroux — CHANGES_REQUESTED** [review]
Looks great! Just a few minor grammar/wording suggestions

File: website/docs/component-standards/interactions/exporting-data.md (hunk 2 above)
Comment (TDH55) [reply website/docs/component-standards/interactions/exporting-data.md:16]: I was going to add something about action arguments but I think it's best to end there.

File: website/docs/component-standards/configuration/package-json.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+- is descriptive
+
+Additionally, when choosing a name for an unpublished library, also choose a name that
+
+- is not already owned by someone else
+- is not spelled in a similar way to another library name
+- will not confuse others about authorship
+
+:::caution
+Once you have published your library you can **NOT** change your library name.
```
[the remark this answers is not in the document]
Comment (adebroux) [reply website/docs/component-standards/configuration/package-json.md]: That's fair! I think we still might want to add a note that you also can't change the description once it's published, just so developers don't get stuck with a placeholder description.

[the remark this answers is not in the document]
Comment (TDH55) [reply website/docs/component-standards/configuration/package-json.md]: It actually looks like we can change the description now so I'm going to leave this as is

**Pull request merged**